lib/phoenix_html/form.ex

```diff
@@ -1300,6 +1301,13 @@ defmodule Phoenix.HTML.Form do
   defp date_value({year, month, day}), do: %{year: year, month: month, day: day}
 
   defp date_value(nil), do: %{year: nil, month: nil, day: nil}
+
+  defp date_value("" <> string) do
+    string
+    |> Date.from_iso8601!()
+    |> date_value
```

Comment:
This `date_value` call isn't required (the tests pass without it), but I figured for consistency it would be a good idea 🤷‍♂️